### PR TITLE
gap: switch more code to offical libgap APIs

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: src/sage/libs/gap/element.pyx [hunk 1]
```diff
@@ -77,21 +78,22 @@ cdef Obj make_gap_matrix(sage_list, gap_ring) except NULL:
...
-    cdef GapElement l = libgap.eval('[]')
+    cdef GapElement l = GAP_NewPlist(0)
     cdef GapElement elem
     cdef GapElement one
     if gap_ring is not None:
         one = <GapElement>gap_ring.One()
     else:
         one = <GapElement>libgap(1)
-    for x in sage_list:
+    for i in range(sage_list):
```
Comment (dcoudert) [inline src/sage/libs/gap/element.pyx]: This won't work. I assume you want `for i, x in enumerate(sage_list):`

**Review by dcoudert — CHANGES_REQUESTED** [review]
Unfortunately, I'm not able to help more than that.

File: src/sage/libs/gap/element.pyx (hunk 1 above)
Comment (fchapoton) [reply src/sage/libs/gap/element.pyx]: I have fixed that.

Comment (fingolfin) [reply src/sage/libs/gap/element.pyx]: Thank you

**Comment by fingolfin** [comment]
There is an error `Cython.Compiler.Errors.CompileError: sage/libs/gap/element.pyx` but I don't see further information about what the problem is. Any hints?

File: src/sage/libs/gap/element.pyx [hunk 2]
```diff
@@ -47,15 +47,15 @@ cdef Obj make_gap_list(sage_list) except NULL:
 
     The list of the elements in ``a`` as a Gap ``Obj``.
     """
-    cdef GapElement l = libgap.eval('[]')
+    cdef GapElement l = GAP_NewPlist(len(sage_list))
```
Comment (fingolfin) [inline src/sage/libs/gap/element.pyx]: Oh I think it is another wrapping issue. I'll submit a potential fix.

**Comment by fingolfin** [comment]
I removed more parts of this PR. I hope the remaining bits will work fine. Perhaps it is best if I wait until this PR is merged, and then re-submit the bits I removed as separate PRs.

**Review by fchapoton — APPROVED** [review]
gut, danke schoen

**Comment by vbraun** [comment]
merge conflict

**Comment by fingolfin** [comment]
rebased

**Review by fchapoton — APPROVED** [review]
ok again

**Pull request merged**